src/containers/HubiiApiHoc/saga.js

```diff
@@ -98,28 +97,57 @@ export function* loadTransactions({ address }, endpoint) {
   }
 }
 
+export function* requestToken() {
+  while (true) { // eslint-disable-line no-constant-condition
+    try {
+      const network = yield select(makeSelectCurrentNetwork());
+      const nahmiiProvider = new nahmii.NahmiiProvider(
+        network.walletApiEndpoint.slice(7, -1),
```

Comment:
We might need to think about the issue where the endpoint could be changed and the length of the string could be changed as well. Maybe we could use regex to extract the string in a safer way.

Reply:
Very good point, will change to regex

Reply:
Or we could isolate that string into a new property in the `network` object, and convert `walletApiEndpoint` into a function that concats the host string property, so no need to do any extractions.